### PR TITLE
Uploads

### PR DESCRIPTION
**Pull request by thessnelson**

### CODE REVIEW

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -4,7 +4,8 @@ class CoursesController < ApplicationController
 
   # GET /courses or /courses.json
   def index
-    @courses_db_result = Course.where(teacher: current_user.email)
+    @courses_db_result = Course.search(params[:search], current_user.email)
+    #@courses_db_result = Course.where(teacher: current_user.email)
```
Comment (kunalvudathu) [inline app/controllers/courses_controller.rb]: remove this comment line

File: app/controllers/students_controller.rb [hunk 2]
```diff
@@ -3,7 +3,8 @@ class StudentsController < ApplicationController
     before_action :set_student, only: %i[ show edit update destroy ]
     # GET /student
     def index
-        @students = Student.where(teacher: current_user.email)
+        @students = Student.search(params[:search], current_user.email)
+        #@students = Student.where(teacher: current_user.email)
```
Comment (kunalvudathu) [inline app/controllers/students_controller.rb]: same here remove this

File: app/controllers/upload_controller.rb [hunk 3]
```diff
@@ -1,61 +1,90 @@
...
+  
+      #if the number of rows in the csv file is equal to the number of images in the zip file, then proceed. Otherwise, throw an error
+      if ((csv.length != 0) && (csv.length == images.length))
+        csv.zip(images).each do |row, image|
+          uuid = SecureRandom.uuid
+  
+          #if row contains all the necessary columns, keep going.
+          if ((row['FirstName']) && (row['LastName']) && (row['Email']) && (row['UIN']) && (row['Section']) && (row['Course']) && (row['Semester']) && (row['Classification']) && (row['Major']) && (row['Notes']))
+            #puts(row)
+            #Rails.logger.info "Collected all student courses #{@student.inspect}"
```
Comment (kunalvudathu) [inline app/controllers/upload_controller.rb]: remove these lines

File: app/controllers/upload_controller.rb [hunk 4]
```diff
@@ -1,61 +1,90 @@
...
-                    lastname:record["LastName"].strip(),
-                    uin: record["UIN"].strip(),
-                    email: record["Email"].strip(),
-                    classification: record["Classification"].strip(),
-                    major: record["Major"].strip(),
-                    notes: record["Notes"].strip(),
-                    teacher: current_user.email
-                )
-                @student.save
+              Rails.logger.info "here all student courses #{@student.inspect}"
```
Comment (kunalvudathu) [inline app/controllers/upload_controller.rb]: this one too

File: app/models/course.rb [hunk 5]
```diff
@@ -1,2 +1,17 @@
 class Course < ApplicationRecord
-end
+    def self.search(search, teacher)
+        if search
+          search_type = Course.find_by(course_name: search, teacher: teacher)
+          Rails.logger.info "Received: #{search_type.inspect}"
```
Comment (kunalvudathu) [inline app/models/course.rb]: remove this print statement

**Review by kunalvudathu — COMMENTED** [review]
Looks great! Just remove unneeded print statements.

**Review by csimpson2018 — APPROVED** [review]

**Review by csimpson2018 — APPROVED** [review]

**Review by rahulshah521 — APPROVED** [review]

**Pull request merged**